JumpscaleLibs/clients/stellar/StellarClient.py

```
@@ -0,0 +1,182 @@
+"""
+Stellar Client
+"""
+
+from Jumpscale import j
+try:
+    from stellar_sdk import Server, Keypair, TransactionBuilder, Network
+    from stellar_sdk.exceptions import BadRequestError
+except (ModuleNotFoundError, ImportError):
+    j.builders.runtime.python3.pip_package_install("stellar_sdk")
+
+try:
+    from stellar_base import Address
+except (ModuleNotFoundError, ImportError):
+    j.builders.runtime.python3.pip_package_install("stellar_base")
+
+from stellar_sdk import Server, Keypair, TransactionBuilder, Network
+from stellar_sdk.exceptions import BadRequestError
+from stellar_base import Address
+
+import requests
+
+JSConfigClient = j.baseclasses.object_config
+
+_HORIZON_NETWORKS = {
+    "TEST": "https://horizon-testnet.stellar.org",
+    "STD": "https://horizon.stellar.org",
+}
+
+class StellarClient(JSConfigClient):
+    """
+    Stellar client object
+    """
+
+    _SCHEMATEXT = """
+        @url = jumpscale.stellar.client
+        name** = "" (S)
+        network** = "STD,TEST" (E)
+        address = (S)
+        secret = (S)
+        """
+
+    def new_keypair(self):
+        """Creates a new Stellar keypair (address and secret)
+        """
+        kp = Keypair.random()
+        self.address = kp.public_key
+        self.secret = kp.secret
+        print("Key: {}".format(kp.secret))
+        print("Address: {}".format(kp.public_key))
+
+    def get_balance(self):
+        """Gets balance for address
+        """
+        address = Address(address=self.address)
+        address.get()
+
+        print('Balances: {}'.format(address.balances))
+
+    def fund_account(self):
+        """Funds a testnet address
+        """
+        if str(self.network) != "TEST":
+            return Exception("You can only fund your account when connected to testnet")
```

I think you want to `raise` instead of return